src/index.ts

```diff
@@ -0,0 +1,59 @@
+import * as core from "@actions/core";
+import { updateCorsConfig } from "./lib/update-cors-cloud";
+import { authenticateCloud } from "./lib/authenticate-cloud";
+
+import { updateClients } from "./lib/update-clients";
+import type { OAuthClientConfig } from "./lib/update-clients";
+
+const AM_URL = core.getInput("AM_URL");
+const username = core.getInput("username");
+const password = core.getInput("password");
+const realm = core.getInput("realm_path");
+const origins = core.getInput("origins"); // this is json as input
+const remove = Boolean(core.getInput("remove_origins"));
+const cookieName = core.getInput("cookie_name");
+const oauthClients = core.getInput("oauthClients"); // this is json also
+
+async function update(): Promise<void> {
+  try {
+    // We first need to authenticate to get an sso token
+    const { tokenId: ssoToken } = await authenticateCloud({
+      AM_URL,
+      realm,
+      username,
+      password,
+    });
+
+    const originsToAdd = JSON.parse(origins)
+      .map((val: { url: string; private: boolean }) => val?.url ?? "")
+      .filter(Boolean);
```

Comment:
I would still like to see a code comment here explaining what this does. Where do the `origins` come from? How can you have an origin without a URL?

Reply:
I don't think you can have an origin without a url, i'm only doing this out of safety for the idea that okteto may not return a perfectly shaped object every time. I'll add a comment